src/internal/serviceenv/service_env.go

```diff
@@ -77,6 +78,14 @@ type NonblockingServiceEnv struct {
 	// returned, then pachd would be unable to start)
 	pachEg errgroup.Group
 
+	// servicePachClient is the "template" service pach client (client connected through the pachd service)
+	// other clients returned by this library are based on. It contains the original GRPC client
+	// connection and has no ctx and therefore no auth credentials or cancellation.
+	servicePachClient *client.APIClient
+	// servicePachEg coordinates the initialization of servicePachClient (see pachdEg)
+	servicePachEg   errgroup.Group
+	servicePachOnce sync.Once
```

Comment:
Either move these into local variables when initializing the proxy listener or expose them with `ServiceEnv` getters (if you have other plans for these) - otherwise I don't see the point of having them here since they're only used in one place.